Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp

```diff
@@ -0,0 +1,228 @@
+#include "Acts/Surfaces/RadialBounds.hpp"
+#include "Acts/Tests/CommonHelpers/CylindricalTrackingGeometry.hpp"
+#include "Acts/Utilities/detail/Grid.hpp"
+
+#include <tuple>
+
+using namespace Acts;
+using namespace Acts::Svg;
+using namespace Acts::Test;
+using namespace Acts::Experimental;
+
+GeometryContext tContext;
+CylindricalTrackingGeometry cGeometry = CylindricalTrackingGeometry(tContext);
+
+IndexedSurfacesConverter::Options generateDrawOptions() {
+  // The converter options
+  IndexedSurfacesConverter::Options isOptions;
+  // Sensitive surface stuyle
+  Style sensitiveStyle;
+  sensitiveStyle.fillColor = {51, 153, 255};
+  sensitiveStyle.fillOpacity = 0.9;
+  sensitiveStyle.highlightColor = {255, 153, 51};
+  sensitiveStyle.highlights = {"onmouseover", "onmouseout"};
+  sensitiveStyle.strokeWidth = 0.5;
+  sensitiveStyle.strokeColor = {0, 0, 0};
+  sensitiveStyle.nSegments = 72u;
+  std::pair<GeometryIdentifier, Style> allSensitives = {GeometryIdentifier(0u),
+                                                        sensitiveStyle};
+
+  // Hierarchy map of styles
+  GeometryHierarchyMap<Style> surfaceStyles({allSensitives});
+  isOptions.surfaceStyles = surfaceStyles;
+
+  // The grid style
+  GridConverter::Options gridOptions;
+  Style gridStyle;
+  gridStyle.fillOpacity = 0.;
+  gridStyle.strokeColor = {0, 0, 255};
+  gridStyle.strokeWidth = 1.;
+  gridStyle.highlightStrokeWidth = 3;
+  gridStyle.highlightStrokeColor = {255, 0, 0};
+  gridOptions.style = gridStyle;
+
+  isOptions.gridOptions = gridOptions;
+  return isOptions;
+};
+
+auto drawOptions = generateDrawOptions();
+
+BOOST_AUTO_TEST_SUITE(DetectorSvg)
+
+BOOST_AUTO_TEST_CASE(RingDisc1D) {
+  // A single ring
+  CylindricalTrackingGeometry::DetectorStore dStore;
+  auto rSurfaces = cGeometry.surfacesRing(dStore, 6.4, 12.4, 36., 0.125, 0.,
+                                          55., 0., 2., 22u);
+
+  IndexedSurfacesGenerator<decltype(rSurfaces)> irSurfaces{
+      rSurfaces, {}, {binPhi}};
+
+  GridAxisGenerators::EqClosed aGenerator{{-M_PI, M_PI}, 44u};
+  PolyhedronReferenceGenerator rGenerator{true, 1};
+
+  auto indexedRing = irSurfaces(tContext, aGenerator, rGenerator);
+  // The displaying
+  auto pIndexedRing = IndexedSurfacesConverter::convert(
+      tContext, rSurfaces, indexedRing, drawOptions);
+  auto pIndexRingView = View::xy(pIndexedRing, "RingDisc1D");
+  toFile({pIndexRingView}, pIndexRingView._id + ".svg");
+}
+
+BOOST_AUTO_TEST_CASE(RingDisc1DWithSupport) {
+  // A single ring
+  CylindricalTrackingGeometry::DetectorStore dStore;
+  auto rSurfaces = cGeometry.surfacesRing(dStore, 6.4, 12.4, 36., 0.125, 0.,
+                                          55., 0., 2., 22u);
+
+  auto rBounds = std::make_shared<RadialBounds>(20., 20.);
+  auto dSurface = Surface::makeShared<DiscSurface>(Transform3::Identity(),
+                                                   std::move(rBounds));
+  rSurfaces.push_back(dSurface.get());
```

Comment:
I am not using smart pointers so much, but why is this a ```shared_ptr``` if all that's needed is a memory address?

Reply:
Surfaces in ACTS must be constructed as shared pointers, all other constructors are deleted/private.

Reply:
Ah, Ok, I didn't know